File: backend/src/main/java/com/hyunprise/backend/domain/coupon/vo/CouponDetail.java
package com.hyunprise.backend.domain.coupon.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CouponDetail {
    private String issuedCouponUUID;
    private String couponName;
    private String couponDescription;
    private Integer discountType;
    private Integer discountAmount;
    private Integer discountLimit;
    private Integer minimumPurchase;
    private String retailerLocation;
    private String termsAndConditions;
    private String usageInstruction;
    private String couponUUID;
    private Integer status;
    private String couponCode;
    private Timestamp issueDate;
    private Timestamp expirationDate;
    private String brandName;
    private Integer equivalentPoint;
}

File: android/app/src/main/java/com/hyunprise/android/api/coupon/vo/CouponDetail.kt
package com.hyunprise.android.api.coupon.vo


import java.sql.Timestamp

data class CouponDetail(
    val issuedCouponUUID: String? = null,
    val couponName: String? = null,
    val couponDescription: String? = null,
    val discountType: Int? = null,
    val discountAmount: Int? = null,
    val discountLimit: Int? = null,
    val minimumPurchase: Int? = null,
    val retailerLocation: String? = null,
    val termsAndConditions: String? = null,
    val usageInstruction: String? = null,
    val couponUUID: String? = null,
    val status: Int? = null,
    val couponCode: String? = null,
    val issueDate: Timestamp? = null,
    val expirationDate: Timestamp? = null,
    val brandName: String? = null,
    val equivalentPoint: Int? = null

)
